### PR TITLE
basic sampling test script, updates to audioset class

### PR DESCRIPTION
**Pull request by cjacoby**
- new Audioset Dataset class
- new method of parsing audioset data
- `auem/sampling.py` to benchmark sampling.
- now ignores files that don't exist instead of failing
- now caches files automatically

### CODE REVIEW

File: auem/data/datasets/audioset.py [hunk 1]
```diff
@@ -90,6 +89,52 @@ def __next__(self):
...
+        if not self._annotations:
+            self._load_annotations()
+
+        item = self._annotations[idx]
+
+        return {
+            "ytid": item[0],
+            "start_seconds": float(item[1].strip()),
+            "end_seconds": float(item[2].strip()),
+            "classes": [self.classes.index(k) for k in item[3].split(",")],
```
Comment (constd) [inline auem/data/datasets/audioset.py:125]: I guess this is a fundamental difference between the two different annotators. Doing it this way basically makes the annotation reader specialized for the only dataset and the task that dataset we've written up to this point. My reasoning about returning everything about the classes in a dictionary in V1 is separation of concerns. This way we'll be able to use the annotation reader in more places without having to rewrite it every time.

File: auem/data/datasets/audioset.py [hunk 2]
```diff
@@ -111,33 +156,41 @@ def __init__(
...
-                            "filepath": datapaths[row["YTID"]],
-                        }
-                    )
+        audioset_path = Path(audioset_path)
+        self.datapaths = {x.stem: x for x in audioset_path.glob("*[m4a|webm]")}
+
+        self.annotations = AudiosetAnnotationReaderV2(
+            audioset_annotations, classes=self.classes
+        )
+        # for row in AudiosetAnnotationReaderV1(f, ontology=self.ontology):
```
Comment (constd) [inline auem/data/datasets/audioset.py:165]: See my comment above, removing this also breaks compatability with V1 :(

File: auem/data/datasets/audioset.py [hunk 3]
```diff
@@ -111,33 +156,41 @@ def __init__(
...
         )
-        audio, sr = librosa.load(sample["filepath"], offset=start, duration=duration)
+
+        try:
+            audio, sr = librosa.load(
+                self.datapaths[sample["ytid"]], offset=start, duration=duration
+            )
+        except KeyError:
+            print(f"**** Missing key: {sample['ytid']} ***")
+            raise
```
Comment (constd) [inline auem/data/datasets/audioset.py:193]: if you raise here, it'll break training!

File: auem/data/datasets/audioset.py [hunk 4]
```diff
@@ -111,33 +156,41 @@ def __init__(
...
 
     def __getitem__(self, idx: int):
         sample = self.annotations[idx]
         start, duration = (
             sample["start_seconds"],
             sample["end_seconds"] - sample["start_seconds"],
         )
-        audio, sr = librosa.load(sample["filepath"], offset=start, duration=duration)
+
+        try:
```
Comment (constd) [inline auem/data/datasets/audioset.py:187]: nice!

File: auem/train.py [hunk 5]
```diff
@@ -18,15 +18,16 @@ def train(cfg: DictConfig) -> None:
...
-    dataset = hydra.utils.get_class(cfg.dataset["class"])(
-        transforms=transforms, **cfg.dataset.params
-    )
 
-    # TODO: there's got to be a better way to do this!
-    train_size = int(0.8 * len(dataset))
-    valid_size = len(dataset) - train_size
-    ds_train, ds_valid = torch.utils.data.random_split(
-        dataset, [train_size, valid_size]
+    ds_train = hydra.utils.get_class(cfg.dataset["class"])(
```
Comment (constd) [inline auem/train.py:22]: this is neat!

File: auem/data/datasets/audioset.py (hunk 1 above)
Comment (cjacoby) [reply auem/data/datasets/audioset.py:153]: I don't _exaclty_ disagree, however I think that taking that approach is too much premature optimization, whereas we just need to get the basic thing done right now; i'm not opposed to adding that back in in the future when it becomes necessary.

File: auem/data/datasets/audioset.py (hunk 2 above)
Comment (cjacoby) [reply auem/data/datasets/audioset.py]: also ditto what i said above

File: auem/data/datasets/audioset.py (hunk 3 above)
Comment (cjacoby) [reply auem/data/datasets/audioset.py:247]: i fixed it

**Pull request merged**